pol-core/bscript/compiler/analyzer/ExpressionEvaluator.cpp

```diff
@@ -61,6 +61,11 @@ void EvaluationVisitor::visit_identifier( Identifier& identifier )
   BObjectRefVec::const_iterator itr = _exec->Globals2->begin(), end = _exec->Globals2->end();
   BObjectRef result;
 
+  if ( !identifier.scoped_name.scope.empty() )
+    throw_invalid_expression();
+
+  auto name = identifier.name();
```

Comment:
I guess I will not fully get the logic. So I have to catch the low hanging fruits.
Could be a const ref to avoid copy

Reply:
Yes, I think that is the case as well. Can be addressed when the expression evaluator gets updated to support scoped identifiers.